### ISSUE TITLE
[FEATURE] Support multi-output for `EstimatorTransformer`

### ISSUE THREAD

**Issue by CarloLepelaars** [koaning/scikit-lego#525]
`EstimatorTransformer` [forces output into a 1-dimensional array](https://github.com/koaning/scikit-lego/blob/307133c07dbc2295292d67fb4bd8d46d653ee28f/sklego/meta/estimator_transformer.py#L41). However, there are quite some use cases where the output of a given estimator outputs a 2-dimensional array (i.e. multi-target). 

Example estimators include [scikit-learn's MultiOutputRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.multioutput.MultiOutputRegressor.html) and [Catboost's multi-output functionality](https://catboost.ai/en/docs/concepts/loss-functions-multiregression).

It looks like the solution is as simple as adding an if-else check to the `transform` method of `EstimatorTransformer`, but there could be additional gotcha's. Happy to contribute this to the library if you are open to generalize `EstimatorTransformer` to multi-output.

Line where multi-out breaks in `EstimatorTransformer` (`.reshape(-1, 1)`):
https://github.com/koaning/scikit-lego/blob/307133c07dbc2295292d67fb4bd8d46d653ee28f/sklego/meta/estimator_transformer.py#L41

**Comment by koaning** [comment]
This sounds like a reasonable feature. What might be the best solution for this? Adding a `multi_output` flag that doesn't reshape at the end?

**Comment by CarloLepelaars** [comment]
Interesting idea! The output might already have the correct shape without reshaping. `test_estimatortransformer::test_shape` is passing even without `reshape(-1, 1)` at the end. 

I wrote an additional test for `EstimatorTransformer` with multitarget (i.e. `test_shape_multitarget`) and debugging it now.

[Pull request](https://github.com/koaning/scikit-lego/pull/526)

**Comment by CarloLepelaars** [comment]
In my test case I specified to pass `random_xy_dataset_multiclf`, but still it seems to pass a single classification dataset. Is `random_xy_dataset_multiclf` actually a single classification dataset or is it passing a different object because of the way the test cases are set up?  

Test case:
https://github.com/CarloLepelaars/scikit-lego/blob/4fe9542fc67a0b9cfa313b49904ea770926628a4/tests/test_meta/test_estimatortransformer.py#L75

**Comment by koaning** [comment]
Let's move all communication on this feature to the PR.

**Comment by koaning** [comment]
PR went in. Will try to release this fix, and some other ones, this week.
